### PR TITLE
MNT: Drop Python 3.7 and update test matrix again

### LINKED ISSUE #175: MNT: Drop Python 3.7 after releasing v0.16.0
As discussed in https://github.com/astropy/sphinx-automodapi/issues/173#issuecomment-1673483548

### PR DESCRIPTION
**Pull request by pllim**
MNT: Drop Python 3.7

TST: Test against Sphinx 7.2 instead of 7.1

Fix #175

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -11,7 +11,7 @@ deps =
     sphinx53: sphinx==5.3.*
     sphinx62: sphinx==6.2.*
     sphinx70: sphinx==7.0.*
-    sphinx71: sphinx==7.1.*
```
Comment (bsipocz) [inline tox.ini:14]: Isn't this considered major release? I would consider keeping this for now.

Comment (pllim) [reply tox.ini:14]: Given all the breakage from yesterday, I guess, LoL. Can I drop "7.0" then? I don't want to keep a super long list of old Sphinx releases.

Comment (bsipocz) [reply tox.ini:14]: I would say wait? It's released back in April, and maybe there are still some <7.1 pins around. I don't expect much failures, but then we also don't run the test suite that often that it should be that big of a deal to cover all cases.

Dropping down to a single version for old major numbers sounds good though.

Comment (pllim) [reply tox.ini:14]: Do you have a preference between 5.3 and 6.2?

Comment (bsipocz) [reply tox.ini:14]: > Wouldn't it make sense to pick up the bug fixes too?

No, I think that would be the same logic of supporting only the latest and greatest version, as that one has the most features and fixes.

My logic here is that the config files state a minimum required version, our task is to make sure that minimum is in fact the minimum and not just aspiration, therefore that minimum should be the one tested.

Comment (pllim) [reply tox.ini:14]: I think I addressed your comments but I am not sure. Please re-review. Thanks!

Comment (bsipocz) [reply tox.ini:14]: Oh, I mean keep one of each of the major versions (e.g. don't drop 5.x.y), but one should be enough, no need for multiple between 5.x and 6.y

File: tox.ini [hunk 2]
```diff
@@ -8,6 +8,7 @@ isolated_build = true
 changedir = .tmp/{envname}
 deps =
     sphinx40: sphinx==4.0.0
```
Comment (bsipocz) [inline tox.ini:10]: you'll hate this bikeshedding, but what about:
Then we can just change that pin depending on the drop schedule, etc. 
```suggestion
    sphinx_oldest: sphinx==4.0.0
```

File: .github/workflows/ci_workflows.yml [hunk 3]
```diff
@@ -30,6 +30,9 @@ jobs:
           - os: ubuntu-latest
             python-version: 3.8
             toxenv: py38-test-sphinx40
```
Comment (bsipocz) [inline .github/workflows/ci_workflows.yml:32]: If you like the other idea
```suggestion
            toxenv: py38-test-sphinx_oldest
```

**Review by bsipocz — APPROVED** [review]
One more inline comment, but it's 🚲 🏠 , so up to you whether to have it or not, the PR is ready to go.

File: tox.ini [hunk 4]
```diff
@@ -1,17 +1,18 @@
 [tox]
-envlist = py{37,38,39,310,311,312}-test-sphinx{40,53,62,70,71,dev}{-cov}{-clocale}
+envlist = py{38,39,310,311,312}-test-sphinx{40,53,62,70,71,72,dev}{-cov}{-clocale}
```
Comment (pllim) [inline tox.ini:2]: Would have to do this too, I guess.
```suggestion
envlist = py{38,39,310,311,312}-test-sphinx{_oldest,53,62,70,71,72,dev}{-cov}{-clocale}
```

**Comment by bsipocz** [comment]
Thanks!

**Pull request merged**